src/back/spv/block.rs

```diff
@@ -1699,13 +1705,30 @@ impl<'w> BlockContext<'w> {
     pub(super) fn write_block(
         &mut self,
         label_id: Word,
-        statements: &[crate::Statement],
+        naga_block: &crate::Block,
         exit: BlockExit,
         loop_context: LoopContext,
+        debug_info: Option<&DebugInfoInner>,
     ) -> Result<(), Error> {
         let mut block = Block::new(label_id);
-
-        for statement in statements {
+        for (statement, span) in naga_block.span_iter() {
+            if let (Some(debug_info), false) = (
+                debug_info,
+                matches!(
+                    statement,
+                    &(Statement::Block(..)
+                        | Statement::If { .. }
+                        | Statement::Switch { .. }
+                        | Statement::Loop { .. })
```

Comment:
Oh...sorry for missing those exceptions.

> Maybe popping the last Line instruction if the instruction about to be pushed is a termination instruction? But the changes required to get that working are quite large...

After read the spec again, I think the only problem is `Loop` statement.
`If` and `Switch` is good to put `OpLine` right ahead because they start by `OpSelectionMerge`
`Loop` is start with a new `OpBranch` (which shouldn't put `OpLine` ahead) and then the`OpLoopMerge` (which we need to put `OpLine`).

Reply:
Right, we only need to omit `OpLine` before termination instructions.

Reply:
> Right, we only need to omit `OpLine` before termination instructions.

sorry, I'am still a bit confused. Isn't that OpLine shouldn't precede termination instructions by the spec?

My current idea is keep exceptions like `Break`,`Continue` , plus `Loop`, and take the `Loop` as special case as inserting `OpLine` before `OpLoopMerge` emits.

Reply:
That sounds right. I used **omit** in my previous message, I guess it can be easily mistaken for **emit** 😅.